include/fmt/format.h

```diff
@@ -967,14 +967,17 @@ FMT_CONSTEXPR20 inline auto count_digits(uint64_t n) -> int {
 template <int BITS, typename UInt>
 FMT_CONSTEXPR auto count_digits(UInt n) -> int {
 #ifdef FMT_BUILTIN_CLZ
-  if (num_bits<UInt>() == 32)
+  if (num_bits<UInt>() == 32) {
```

Comment:
Let's wrap the condition in `const_check` instead.

Reply:
```c++
template <int BITS, typename UInt>
FMT_CONSTEXPR auto count_digits(UInt n) -> int {
#ifdef FMT_BUILTIN_CLZ
  if (const_check(num_bits<UInt>() == 32))
    return (FMT_BUILTIN_CLZ(static_cast<uint32_t>(n) | 1) ^ 31) / BITS + 1;
#endif
  int num_digits = 0;
```
doesn't squash the warning.

Reply:
That's unfortunate but I think the suppression with extra blocks is too messy. Is there another way to do this?

Reply:
eafdd3eb is _another_ way -- I wouldn't say it's a particularly **nice** way...

Reply:
Yeah, this is clearly worse =(. Does moving the definition of `num_digits` to the beginning of the function suppress the warning?

```c++
  int num_digits = 0;
#ifdef FMT_BUILTIN_CLZ
  if (const_check(num_bits<UInt>() == 32))
    return (FMT_BUILTIN_CLZ(static_cast<uint32_t>(n) | 1) ^ 31) / BITS + 1;
#endif
  ```

Reply:
Nice idea, but that just gives
```
warning #128-D: loop is not reachable
```
on the `do {` line. And I would be a little worried about other compilers giving an unused variable warning too...

I pushed yet another attempt, with an immediately executed lambda, which seems to work...

Reply:
This looks better but please fix the CI issues.